src/System.ComponentModel.EventBasedAsync/tests/AsyncOperationFinalizerTests.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading;
+using Xunit;
+
+namespace System.ComponentModel.Tests
+{
+    public class AsyncOperationFinalizerTests : RemoteExecutorTestBase
+    {
+        [Fact]
+        public void Finalizer_GetViaReflection_IsPresentForCompatability()
+        {
+            MethodInfo finalizer = typeof(AsyncOperation).GetMethod("Finalize", BindingFlags.NonPublic | BindingFlags.Instance);
+            Assert.NotNull(finalizer);
+        }
```

Comment:
It'd be better to actually test the functionality of the finalizer rather than testing for its presence via reflection and invoking it explicitly as is done below.  You could do this by creating your own sync ctx that the finalizer will call back to.